integrations/sonarqube/main.py

```diff
@@ -18,64 +25,97 @@ def init_sonar_client() -> SonarQubeClient:
     )
 
 
+def produce_component_params(
+    client: SonarQubeClient, selector: Any, initial_params: dict[str, Any] = {}
+) -> dict[str, Any]:
+    component_query_params: dict[str, Any] = {}
+    if client.organization_id:
+        component_query_params["organization"] = client.organization_id
+
+    ## Handle query_params based on environment
+    if client.is_onpremise:
+        if initial_params:
+            component_query_params.update(initial_params)
+        elif event.resource_config:
+            # This might be called from places where event.resource_config is not set
+            # like on_start() when creating webhooks
+
+            selector = cast(CustomSelector, event.resource_config.selector)
+            component_query_params.update(selector.generate_request_params())
+    return component_query_params
+
+
 sonar_client = init_sonar_client()
 
 
 @ocean.on_resync(ObjectKind.PROJECTS)
 async def on_project_resync(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
     logger.info(f"Listing Sonarqube resource: {kind}")
-    fetched_projects = False
-    async for project_list in sonar_client.get_all_projects():
-        yield project_list
-        fetched_projects = True
-
-    if not fetched_projects:
-        logger.error("No projects found in Sonarqube")
-        raise RuntimeError(
-            "No projects found in Sonarqube, failing the resync to avoid data loss"
-        )
+
+    selector = cast(SonarQubeProjectResourceConfig, event.resource_config).selector
+    sonar_client.metrics = selector.metrics
+
+    component_params = produce_component_params(sonar_client, selector)
+
+    async for projects in sonar_client.get_components(query_params=component_params):
+        logger.info(f"Received project batch of size: {len(projects)}")
+        yield projects
+
+
+@ocean.on_resync(ObjectKind.GA_PROJECTS)
+async def on_ga_project_resync(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
+    logger.info(f"Listing Sonarqube resource: {kind}")
+
+    selector = cast(SonarQubeGAProjectResourceConfig, event.resource_config).selector
+    sonar_client.metrics = selector.metrics
+
+    async for projects in sonar_client.get_components(
+        query_params=selector.generate_request_params()
+    ):
```

Comment:
why is the ga_project kind using the get_components() which calls the internal API instead of the get_projects() which calls the GA API?